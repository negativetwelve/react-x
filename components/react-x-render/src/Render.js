// Libraries
import React from 'react';
import PropTypes from 'prop-types';


const Render = ({if: ifCondition, unless, children}) => {
  if (ifCondition && !unless && children) {
    return React.Children.only(children);
  } else {
    return null;
  }
};

// --------------------------------------------------
// Props
// --------------------------------------------------
Render.propTypes = {
  if: PropTypes.bool,
  unless: PropTypes.bool,
};

Render.defaultProps = {
  if: true,
  unless: false,
};


export default Render;
