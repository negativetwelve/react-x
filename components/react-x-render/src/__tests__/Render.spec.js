// Modules
import React from 'react';
import Render from '../Render';


const Text = ({children}) => <div>{children}</div>;

/* eslint-disable no-undef */
describe('Render', () => {
  context('with default props', () => {
    action('render', () => (
      <Render>
        <Text>{'Something rendered'}</Text>
      </Render>
    ));

    itRenders(() => render());
  });

  context('with if prop', () => {
    set('ifCondition', () => false);
    action('render', () => (
      <Render if={ifCondition}>
        <Text>{'Something rendered'}</Text>
      </Render>
    ));

    context('with ifCondition set to true', () => {
      set('ifCondition', () => true);
      itRenders(() => render());
    });

    context('with ifCondition set to false', () => {
      set('ifCondition', () => false);
      itRenders(() => render());
    });
  });

  context('with unless prop', () => {
    set('unlessCondition', () => true);
    action('render', () => (
      <Render unless={unlessCondition}>
        <Text>{'Something rendered'}</Text>
      </Render>
    ));

    context('with unlessCondition set to true', () => {
      set('unlessCondition', () => true);
      itRenders(() => render());
    });

    context('with unlessCondition set to false', () => {
      set('unlessCondition', () => false);
      itRenders(() => render());
    });
  });
});
