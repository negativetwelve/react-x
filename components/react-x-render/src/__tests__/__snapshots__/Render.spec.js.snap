// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render with default props renders snapshot 1`] = `
<div>
  Something rendered
</div>
`;

exports[`Render with if prop with ifCondition set to false renders snapshot 1`] = `null`;

exports[`Render with if prop with ifCondition set to true renders snapshot 1`] = `
<div>
  Something rendered
</div>
`;

exports[`Render with unless prop with unlessCondition set to false renders snapshot 1`] = `
<div>
  Something rendered
</div>
`;

exports[`Render with unless prop with unlessCondition set to true renders snapshot 1`] = `null`;
